specs/xaml-resource-references-tracing-spec.md

```diff
@@ -0,0 +1,189 @@
+which makes it difficult or, more often, outright impossible to debug.
+
+This spec describes a new 
+API on the `DebugSettings` class that developers can use to access more detailed information about 
+the resource reference lookup failure.
+The new APIs here are patterned after the existing
+[DebugSettings.BindingFailed](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.UI.Xaml.DebugSettings.BindingFailed)
+and
+[DebugSettings.IsBindingTracingEnabled](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.UI.Xaml.DebugSettings.IsBindingTracingEnabled)
+APIs.
+
+
+# API Pages
+
+_(Updates to the 
+[DebugSettings](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.UI.Xaml.DebugSettings)
+class)_
+
+## DebugSettings.XamlResourceReferenceFailed Event
+
+Occurs when a
+[XAML resource reference](https://learn.microsoft.com/en-us/windows/apps/design/style/xaml-resource-dictionary)
+cannot be resolved.
+
+_Spec note: This API is similar to the `DebugSettings.BindingFailed` event. Like XAML resource references, 
+Bindings cannot be modeled solely through static evaluation, e.g. at compile-time, and so a run-time 
+event when a failure occurs is the best approach for providing developers with a means of determining the 
+root cause._
+
+_Spec note: is there a version of the 'ResourceDictionary and XAML resource references' article that 
+links to the Windows App SDK documentation for relevant APIs?_
+
+```c#
+public event TypedEventHandler<DebugSettings,XamlResourceReferenceFailedEventArgs> 
+XamlResourceReferenceFailed
+
+```
+
+### Remarks
+
+`IsXamlResourceReferenceTracingEnabled` must be `true` in order for this event to be raised.
+
+Error information is also logged to the native debug output,
+so attaching a `XamlResourceReferenceFailed` handler yourself is an advanced scenario for 
+getting the raw message programmatically.
+
+## DebugSettings.IsXamlResourceReferenceTracingEnabled Property
+
+Gets or sets a value indicating that when a XAML resource reference error occurs,
+the `XamlResourceReferenceFailed` event should be raised,
+and error information should be logged in the native debug output.
+
+```c# 
+public bool IsXamlResourceReferenceTracingEnabled { get; set; }
+```
+
+### Remarks
+
+This property is `true` by default. When XAML resource reference tracing is enabled and you 
+run your app with the native debugger attached, any  XAML resource reference errors appear 
+in the **Output** window in Microsoft Visual Studio.
+
+
+## XamlResourceReferenceFailedEventArgs Class
+
+Provides event data for the `DebugSettings.XamlResourceReferenceFailed` event.
+
+```c#
+public sealed class XamlResourceReferenceFailedEventArgs
+```
+
+### Remarks
+
+`XamlResourceReferenceFailedEventArgs` is used for debugging XAML resource references. Wire the event 
+handler using `DebugSettings`, and use this data class as the result in your handler. You'll mainly be 
```

Comment:
This data type is not the result. It is an input to the handler.
```suggestion
handler using `DebugSettings`. It will receive a reference to this class. You'll mainly be 
```